examples/IB/explicit/ex7/example.cpp

```diff
@@ -391,95 +419,81 @@ main(int argc, char* argv[])
 
             IBTK_MPI::barrier();
             // initialize vector to store the nodal coordinates on each processor
-            std::vector<double> local_coordinates(num_nodes * NDIM);
-            int k = 0;
-
+            std::vector<Lag_Coords> local_coordinates;
             for (const auto& node : local_nodes)
             {
                 const int petsc_idx = node->getLocalPETScIndex();
+                const int lag_idx = node->getLagrangianIndex();
+                // put the lag idx as the first entry of the local_coordinates vector since the petsc indices may change
+                // throughout a computation
                 Eigen::Map<Vector3d> X(&x_vals[petsc_idx * NDIM]); // X is a vector containing the local coordinates
-                for (int m = 0; m < NDIM; m++)
-                {
-                    local_coordinates[NDIM * k + m] = X[m];
-                }
 
-                k = k + 1;
+                local_coordinates.push_back({ lag_idx, X[0], X[1], X[2] });
```

Comment:
I'm surprised that this works - I don't think its valid in plain C++11. Could you add a constructor to `Lag_Coords` and call that instead?